api/external/applications/applications.proto

```diff
@@ -212,3 +212,30 @@ enum HealthStatus {
   CRITICAL = 2;
   UNKNOWN = 3;
 }
+
+
+message Wrapper {
+	// From the proto compiler:
+	// Note that enum values use C++ scoping rules, meaning that enum values are
+	// siblings of their type, not children of it.
+	//
+	// So we need to nest this one extra level
+	enum WrappedMessageType {
+		UNKNOWN = 0;
+		EXAMPLE_ONE = 1;
+		EXAMPLE_TWO = 2;
+	}
+
+  WrappedMessageType type = 1;
```

Comment:
Making this an enum I think gives us the lowest possible overhead in terms of additional bytes on the wire. If you run the test code, it prints:
```
inner msg: [8 23 16 42]
inner msg len: 4
data: [8 1 18 4 8 23 16 42]
data len: 8
```

So we only added 4 bytes

Reply:
For further discussion, in the last load test, we had about 600 messages per second, so this would add 4 * 600 == 2400 bps, which should be pretty much insignificant.